anvio/filesnpaths.py

```diff
@@ -368,3 +368,192 @@ def get_name_from_file_path(file_path, postfix_separator="."):
     splits = os.path.basename(os.path.abspath(file_path)).split(postfix_separator)
 
     return splits[0] if len(splits) in [1, 2] else '.'.join(splits[:-1])
+
+
+class AppendableFile:
+    """A file class for repeatedly opening the same file and appending data to it.
+
+    Parameters
+    ==========
+    file_path : string
+        the path to the file
+    append_type : data type, None
+        If you provide a data type here, then only data of that type will be allowed to be appended to the file
+    fail_if_file_exists : boolean
+        if True, throw an error upon initialization if the file already exists. If False,
+        start appending to the end of the file.
```

Comment:
I think `fail_if_file_exists` should be `False` by default. I think this because the class name is `AppendableFile`, so programmers who use it are expecting to append to a file that already exists, not be complained at because the file exists.

Reply:
Okay, this makes sense. I will change it.